app/src/main/java/chat/rocket/android/profile/presentation/ProfilePresenter.kt

```diff
@@ -0,0 +1,66 @@
+package chat.rocket.android.profile.presentation
+
+import chat.rocket.android.core.lifecycle.CancelStrategy
+import chat.rocket.android.helper.UrlHelper
+import chat.rocket.android.server.domain.GetCurrentServerInteractor
+import chat.rocket.android.server.infraestructure.RocketChatClientFactory
+import chat.rocket.android.util.launchUI
+import chat.rocket.common.RocketChatException
+import chat.rocket.common.util.ifNull
+import chat.rocket.core.RocketChatClient
+import chat.rocket.core.internal.rest.me
+import chat.rocket.core.internal.rest.updateProfile
+import javax.inject.Inject
+
+class ProfilePresenter @Inject constructor (private val view: ProfileView,
+                                            private val strategy: CancelStrategy,
+                                            serverInteractor: GetCurrentServerInteractor,
+                                            factory: RocketChatClientFactory) {
+    private val serverUrl = serverInteractor.get()!!
+    private val client: RocketChatClient = factory.create(serverUrl)
+    private lateinit var myselfId: String
+
+    fun loadUserProfile() {
+        launchUI(strategy) {
+            view.showLoading()
+            try {
+                val myself = client.me()
+                myselfId = myself.id
+                val avatarUrl = UrlHelper.getAvatarUrl(serverUrl, myself.username!!)
+                view.showProfile(avatarUrl, myself.name!!, myself.username!!, myself.emails?.get(0)?.address!!)
+            } catch (exception: RocketChatException) {
+                exception.message?.let {
+                    view.showMessage(it)
+                }.ifNull {
+                    view.showGenericErrorMessage()
+                }
+            } finally {
+                view.hideLoading()
+            }
+        }
+    }
+
+    fun updateUserProfile(email: String, name: String, username: String) {
+        launchUI(strategy) {
+            view.showLoading()
+            try {
+                val user = client.updateProfile(myselfId, email, name, username)
```

Comment:
We need to check it later with @luciofm too, but I do think that we would or at least should fall throw to catch to the `RocketChatException` on all calls.

Reply:
@leonardoaramaki you are right, we will need to add exception handling code (it can be done with a default handler for the coroutine), but for now, crashing is good because we got to fix the real problem instead of just hiding it.